src/unix/mod.rs

```diff
@@ -206,9 +206,19 @@ pub const PRIO_MAX: ::c_int = 20;
 cfg_if! {
     if #[cfg(dox)] {
         // on dox builds don't pull in anything
-    } else if #[cfg(all(not(stdbuild), feature = "use_std"))] {
+    } else if #[cfg(all(not(stdbuild),
+                        feature = "use_std",
+                        not(any(target_os = "macos",
+                                target_os = "ios")
+                        )))] {
         // cargo build, don't pull in anything extra as the libstd  dep
         // already pulls in all libs.
+    } else if #[cfg(all(not(stdbuild), feature = "use_std"))] {
+        // except on macOS and iOS, where we must link with lib resolv
+        // for res_init, despite libsystem_info including it:
+        // http://blog.achernya.com/2013/03/os-x-has-silly-libsystem.html
+        #[link(name = "resolv")]
```

Comment:
So we should simply not link with `resolv` on macOS/iOS, both on `stdbuild` and `not(stdbuild)` , and disable the test for `res_init if apple`?